curriculum/challenges/english/25-front-end-development/workshop-email-simulator/6853f4ded9f84896ff182dd8.md

```diff
@@ -0,0 +1,120 @@
+---
+id: 6853f4ded9f84896ff182dd8
+title: Step 19
+challengeType: 20
+dashedName: step-19
+---
+
+# --description--
+
+After the empty inbox check, add a print statement with the message `'\nYour Emails:'`. After that, iterate over all emails in the inbox using a `for` loop. Use enumeration to number the emails starting from `1`. Unpack the enumeration into `i` (the index) and `email` (the email object). Inside the loop, print each email using the `__str__` method you defined earlier, formatted with its number like this:
```

Comment:
Directly mentioning `__str__` here can be misleading. People might think they should actually call `__str__`. I think so far we've never clearly stated how `__str__` is called under the hood. So this might need some explanations.

```suggestion
After the empty inbox check, print the message `'\nYour Emails:'`. After that, iterate over all emails in the inbox using a `for` loop. Use enumeration to number the emails starting from `1`. Use `i` (the index) and `email` (the email object) as the iteration variables.

Inside the loop, print an f-string made by the iteration index followed by a `.`, then a space and the string representation of the email object like this:
```